Core.IntegrationTests/GitBackedTestBase.cs

```diff
@@ -97,14 +97,18 @@ protected static string ExecuteGitCommandWithOutput (string argument)
     psi.RedirectStandardError = true;
 
     using var command = Process.Start(psi);
+
+    var errorMessage = command!.StandardError.ReadToEnd();
+    var output = command.StandardOutput.ReadToEnd();
```

Comment:
I think this logic might have the same issue as before. The problem comes from the fact that we the streams have a limited capacity and will block the started process if full. I think we have the same problem with this code if the output stream is full. In such a case we would block in line 101 instead?

To solve this we need to read both streams async, wait for both to finish and then collect the results.